### PR TITLE
external/error_report: Fix timestamp, use MONOTONIC instead of REALTIME

### PR DESCRIPTION
**Pull request by dr-venkman**
At present, error report module uses REALTIME clock for getting timestamps. The recorded values can be affected by a change of time base during runtime (on response from NTP server, for example). As a fix, we choose relative timestamps instead.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9a29aa128146c46a1d3fa511ec4d2e3b26c9c674] -

**Comment by seinfra** [comment]
Target : [9a29aa128146c46a1d3fa511ec4d2e3b26c9c674] -

**Comment by seinfra** [comment]
Target : [6a9458e475fa3bcd5688c53b649c9eb74a541b2a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6a9458e475fa3bcd5688c53b649c9eb74a541b2a] - Code Rule Check (C++) OK.

File: external/error_report/error_report.c [hunk 1]
```diff
@@ -80,7 +81,16 @@ err_status_t error_report_deinit(void)
 error_data_t *error_report_data_create(int module_id, int error_code, uint32_t pc_value)
 {
 	struct timeval ts;
-	gettimeofday(&ts, NULL);
+	struct timespec tv;
+	int retval = clock_gettime(CLOCK_MONOTONIC, &tv);
```
Comment (sunghan-chang) [inline external/error_report/error_report.c]: When CONFIG_CLOCK_MONOTONIC is enabled, you can use this mode. Did you make a dependency between err_report and clock_monotonic? If you didn't, you should do it.

Comment (dr-venkman) [reply external/error_report/error_report.c]: Added a check for MONOTONIC before use. Thanks for spotting this.

**Comment by seinfra** [comment]
Target : [a4178e357cb8f0bae9fa603be16a08abcf0034e3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a4178e357cb8f0bae9fa603be16a08abcf0034e3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [53da37ea6fa7b4c7eb674d70b9ac0742cf8d8420] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [53da37ea6fa7b4c7eb674d70b9ac0742cf8d8420] - Code Rule Check (C++) OK.

**Pull request merged**